src/main/java/jewellery/inventory/service/ResourceInOrganizationService.java

```diff
@@ -0,0 +1,208 @@
+
+    ResourceInOrganization resourceInOrganization =
+        addResourceToOrganization(
+            organization,
+            resource,
+            resourceInOrganizationRequestDto.getQuantity(),
+            resourceInOrganizationRequestDto.getDealPrice());
+
+    return resourceInOrganizationMapper.toResourceInOrganizationResponse(resourceInOrganization);
+  }
+
+  @Transactional
+  @LogUpdateEvent(eventType = ORGANIZATION_REMOVE_RESOURCE_QUANTITY)
+  public ResourcesInOrganizationResponseDto removeQuantityFromResource(
+      UUID organizationId, UUID resourceId, BigDecimal quantity) {
+    return removeQuantityFromResourceNoLog(organizationId, resourceId, quantity);
+  }
+
+  public ResourcesInOrganizationResponseDto getAllResourcesFromOrganization(UUID organizationId) {
+    Organization organization = organizationService.getOrganization(organizationId);
+    userInOrganizationService.validateUserInOrganization(organization);
+
+    return resourceInOrganizationMapper.toResourceInOrganizationResponse(organization);
+  }
+
+  @Override
+  public Object fetchEntity(Object... ids) {
+    if (ids != null && ids.length > 0) {
+      if (ids[0] instanceof ResourceInOrganizationRequestDto resourceInOrganizationRequestDto) {
+
+        return getResourceInOrganizationResponse(
+            resourceInOrganizationRequestDto.getOrganizationId(),
+            resourceInOrganizationRequestDto.getResourceId());
+      } else {
+        return getResourceInOrganizationResponse((UUID) ids[0], (UUID) ids[1]);
+      }
+    }
+    return null;
+  }
+
+  private ResourcesInOrganizationResponseDto getResourceInOrganizationResponse(
+      UUID organizationId, UUID resourceId) {
+    Organization organization = organizationService.getOrganization(organizationId);
+    Resource resource = resourceService.getResourceById(resourceId);
+    ResourceInOrganization resourceInOrganization =
+        getResourceInOrganization(organization, resource);
+    if (resourceInOrganization != null) {
+      return resourceInOrganizationMapper.toResourceInOrganizationResponse(resourceInOrganization);
+    }
+    return null;
+  }
+
+  private ResourcesInOrganizationResponseDto removeQuantityFromResourceNoLog(
+      UUID organizationId, UUID resourceId, BigDecimal quantity) {
+    Organization organization = organizationService.getOrganization(organizationId);
+
+    throwExceptionWhenQuantityToRemoveIsNegative(quantity);
+
+    userInOrganizationService.validateCurrentUserPermission(
+        organization, OrganizationPermission.REMOVE_RESOURCE_QUANTITY);
+
+    ResourceInOrganization resourceInOrganization =
+        findResourceInOrganizationOrThrow(organization, resourceId);
+    removeQuantityFromResource(resourceInOrganization, quantity);
+
+    if (resourceInOrganization != null) {
+      return resourceInOrganizationMapper.toResourceInOrganizationResponse(resourceInOrganization);
+    }
+    return new ResourcesInOrganizationResponseDto();
```

Comment:
I think we should move the null check handling in the mapper. It makes sense the mapper is able to handle null objects. Then this becomes
```suggestion

    return resourceInOrganizationMapper.toResourceInOrganizationResponse(resourceInOrganization);
 
```

Reply:
Done! I moved thiss check to the mapper